src/Sandbox/SourcePolicyInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Sandbox;
+
+use Twig\Source;
+
+/**
+ * Interface for a class that can optionally enable the sandbox mode based on a template's Twig\Source.
+ *
+ * @author Yaakov Saxon
+ */
+interface SourcePolicyInterface
+{
+    public function enableSandbox(?Source $source): bool;
```

Comment:
should we even support null in this interface ? I don't know how an implementation would take a decision in this case. We should rather handle that case in the SandboxExtension, calling the source policy only when we have a source (which should always be the case in calls done in compiled templates rather than custom callers anyway)

Reply:
I wasn't entirely sure under what circumstances Source could actually be null, but my thought was to allow the implementer to decide how to handle the null cases - essentially a question of whether to whitelist acceptable Sources (and block nulls) or blacklist unacceptable ones (and accept nulls)

Reply:
But in any case I just added commit da79bdc1f5aeb8135701271049aab8b9b1f02309 as an implementation of what you said

Reply:
(Post rebase, that's now commit cbbf38b7db8b768eec0aca1928aef2f2df80de64)